crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -909,33 +909,42 @@ impl<'src> Parser<'src> {
         }
     }
 
+    /// Parses a boolean operation expression.
+    ///
+    /// Note that the boolean `not` operator is parsed as a unary operator and
+    /// not as a boolean operation.
+    ///
+    /// # Panics
+    ///
+    /// If the parser isn't positioned at a `or` or `and` token.
+    ///
+    /// See: <https://docs.python.org/3/reference/expressions.html#boolean-operations>
     fn parse_bool_operation_expression(
         &mut self,
         lhs: Expr,
         start: TextSize,
-        op: TokenKind,
-        op_bp: Precedence,
+        operator_token: TokenKind,
+        operator_binding_power: Precedence,
     ) -> ast::ExprBoolOp {
```

Comment:
Nit: I think we could do something similar here as I pointed out in the compare expression PR to avoid having to pass `operator_token`, `operator_binding_power`.

1. Implement `From<BoolOp> for TokenKind`
1. Implement `From<BoolOp> for Precedence`
1. Implement `as_bool_operator` on `TokenKind` that retuurns `Option<BoolOp>`
1. Change the signature to accept a `BoolOp` instead of `operator_token` and `operator_binding_power`. Call into the (inlineable) `From` implementations to get the `TokenKind` and `Precedence`


Nit: I would rename `operator_binding_power` to `operator_precedence` or rename `Precedence` to `BindingPower` (although I think the term Precedence is more common)